run-cf-tests.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+set -e
+set -u
+#set -x
+set -o pipefail
+
+### This script runs the Checker Framework's tests that depend on
+### dljc (this project). It is intended to be used as part of a CI integration test.
+### This script must be run from the top-level directory
+### It performs these actions:
+### 1. fetches plume-scripts
+### 2. uses git-clone-related to clone either the Checker Framework master version or the current user's
+###    fork of the CF on the same branch, if it exists
+### 3. re-clones this repo/branch in the clone of the CF at its checker/bin/.do-like-javac directory
+### 4. runs the CF tests that use dljc
+
+### get plume-scripts
+
+if [ -d /tmp/"$USER"/plume-scripts ] ; then
+  git -C /tmp/"$USER"/plume-scripts pull -q > /dev/null 2>&1
+else
+  mkdir -p /tmp/"$USER" && git -C /tmp/"$USER" clone --depth 1 -q https://github.com/plume-lib/plume-scripts.git
+fi
+
+### clone the CF
+
+/tmp/"$USER"/plume-scripts/git-clone-related typetools checker-framework /tmp/"$USER"/checker-framework
+export CHECKERFRAMEWORK=/tmp/"$USER"/checker-framework
+
+### setup the CF to use this repo/branch of dljc
+
+rm -rf /tmp/"$USER"/checker-framework/checker/bin/.do-like-javac
+
+# I know we already have this---we're in a copy of it right now!---but I want to make sure that
```

Comment:
This works, but it means that running this script produces a different file system configuration than running from the `checker-framework` repository.  That could lead to inconsistent test results and confusion.

I opened a pull request to do the work in the Checker Framework:
https://github.com/typetools/checker-framework/pull/4361

That pull request leads to shorter code in the Checker Framework, shorter code here (step 3 is not needed), and (most importantly) it is less error-prone and less likely to lead to inconsistent results.

Reply:
This approach still presents a problem: suppose that I create a new PR to `do-like-javac` to merge some branch `foo`. We want to run `do-like-javac`'s CI tests (i.e. the ones added by this PR) on that new PR. Suppose further that no modifications were necessary in the Checker Framework, so there is no corresponding `foo` Checker Framework branch. Then, your proposal leads to the following scenario:
1. CI checks out the `foo` branch of `do-like-javac` and executes this script
2. `git-clone-related` from this script checks out the `master` branch of `typetools/checker-framework`, because there is no `foo` branch of the Checker Framework
3. in the Checker Framework test scripts, `git-clone-related` checks out the `master` branch of `do-like-javac` into `checker/bin/.do-like-javac`, because the Checker Framework repository is on the `master` branch
4. the tests are run using the `master` branch of `do-like-javac` rather than the intended `foo` branch

Did I miss something in `git-clone-related`'s documentation that makes this scenario not happen? If not, I think I can solve the problem by simply checking in the CF clone that it is on the `foo` branch after running `git-clone-related` from this script, and if not create and checkout a new `foo` branch of the CF locally before invoking the relevant script.